### PR TITLE
[AMDGPU] Add GFX12 __builtin_amdgcn_s_sleep_var

### PR DESCRIPTION
**Pull request by jayfoad**

### CODE REVIEW

File: clang/test/CodeGenOpenCL/builtins-amdgcn-gfx12.cl [hunk 1]
```diff
@@ -5,6 +5,21 @@
...
+// CHECK-NEXT:    call void @llvm.amdgcn.s.sleep.var(i32 [[TMP0]])
+// CHECK-NEXT:    call void @llvm.amdgcn.s.sleep.var(i32 15)
+// CHECK-NEXT:    ret void
+//
+void test_s_sleep_var(int d)
+{
+  __builtin_amdgcn_s_sleep_var(d);
+  __builtin_amdgcn_s_sleep_var(15);
+}
+
```
Comment (arsenm) [inline clang/test/CodeGenOpenCL/builtins-amdgcn-gfx12.cl:22]: Should also add the error test to make sure it's rejected on not-gfx12 like for other builtins

Comment (jayfoad) [reply clang/test/CodeGenOpenCL/builtins-amdgcn-gfx12.cl:22]: Done.

File: clang/test/CodeGenOpenCL/builtins-amdgcn-gfx11-param-err.cl [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+// RUN: %clang_cc1 -triple amdgcn-unknown-unknown -target-cpu gfx1100 -verify -S -emit-llvm -o - %s
+
+typedef unsigned int uint;
+typedef uint uint2 __attribute__((ext_vector_type(2)));
+typedef uint uint4 __attribute__((ext_vector_type(4)));
+
+kernel void builtins_amdgcn_bvh_err(global uint2* out, uint addr, uint data, uint4 data1, uint offset) {
+  *out = __builtin_amdgcn_ds_bvh_stack_rtn(addr, data, data1, offset); // expected-error {{'__builtin_amdgcn_ds_bvh_stack_rtn' must be a constant integer}}
+}
```
Comment (arsenm) [inline clang/test/CodeGenOpenCL/builtins-amdgcn-gfx11-param-err.cl:11]: Unrelated test added?

Comment (arsenm) [reply clang/test/CodeGenOpenCL/builtins-amdgcn-gfx11-param-err.cl:11]: Nevermind, the test names are a bit confusing

**Review by arsenm — APPROVED** [review]

**Pull request merged**